src/main/java/com/nanalysis/sharepoint/SharepointClient.java

```diff
@@ -58,49 +55,13 @@ public SharepointClient(String baseUrl, String site) {
         this.siteUrl = baseUrl + "/sites/" + site;
     }
 
-    public void authenticate(String username, String password)
+    public void authenticateWithUserCredentials(String username, String password)
             throws IOException, InterruptedException, XPathExpressionException, ParserConfigurationException, SAXException {
-        String token = getSecurityToken(username, password);
-        signin(token);
-        this.token = fetchBearerToken();
+        this.token = new UserPasswordAuthenticator(httpClient, baseUrl).authenticate(username, password);
     }
 
-    private String getSecurityToken(String username, String password) throws IOException, InterruptedException,
-            ParserConfigurationException, SAXException, XPathExpressionException {
```

Comment:
Is this method using security token not going to be used anymore?

Reply:
Unless the IT team changes it mind, but I decided to keep it, at least for reference.